content/posts/ytnapariparty.md

```diff
@@ -0,0 +1,128 @@
+---
+title: "A yt-napari party"
+date: 2021-01-12T18:22:25-06:00
+author: Chris Havlin
+cover: /img/ytnapari/ytnapari_enzo3d.png
+categories:
+  - Tutorial
+tags:
+  - Community
+---
+
+Interactive visualization of *yt* data using *napari*? 
+
+<!--more-->
+
+A while back I was playing around with using *napari* to visualize data loaded from *yt* and with the launch of the awesome new *yt* blog, I thought that it's about time to brush off that work and share it! The basic goal was to use *yt* to load a dataset and then generate a series of fixed resolution buffers that get passed off to *napari* for interactive visualization. It's a fun and simple way of adding some more interactive analysis to your *yt* workflow and might serve as a basis for writing a *napari* plugin that uses *yt* as a backend for IO and analysis! 
+
+To whet your appetite, here's a screen capture of interactive visualization of the Enzo64 sample dataset:
+
+{{< youtube id="HhoaXgMNOz0" autoplay="true" title="Interactive visualization using yt and napari">}}
+
+In this post, I'll just hit the highlights of the the full notebook, which you can check out [here](https://nbviewer.jupyter.org/github/chrishavlin/yt_scratch/blob/master/notebooks/yt_napari_part_1.ipynb). 
+
+## Visualizing a *yt* image stack
+
+So in general, using *napari* to visualize a 3D array of data is as simple as:
+
+```python
+viewer = napari.view_image(image_stack)
+```
+
+where `image_stack` is an array object. If we're OK keeping our `image_stack` in memory, we can just sample our *yt* dataset with a [fixed resolution buffer](https://yt-project.org/doc/examining/low_level_inspection.html#examining-image-data-in-a-fixed-resolution-array) and hand that off to *napari*! 
+
+For example, the following code loads up one of *yt*'s sample datasets and pulls out a fixed resolution array containing the log of the density field:
+
+```python
+import yt 
+import numpy as np 
+
+ds = yt.load("IsolatedGalaxy/galaxy0030/galaxy0030") 
+cg = ds.r[::300j,::300j,::300j]
+dens_log = np.log10(cg['density'])
+```
+
+So to use *napari* to visualize our `dens_log` array, we just add it to a viewer. The *napari* GUI uses Qt for rendering, so the exact syntax depends a bit on your python session. In a standard python shell, you can use the `napari.gui_qt()` context:
+
+```python
+import napari
+with napari.gui_qt(): 
+    viewer = napari.Viewer()
+    viewer.add_image(dens_log)
+```
+
+Or if in an IPython session (or Jupyter notebook), use the `%gui qt` magic command:
+
+```
```

Comment:
do you want this to have python formatting or is intentionally not highlighted?

Reply:
yes, python formatting! will add it.